File: clojure/word-count/src/word_count.clj
(ns word-count
  (:use [clojure.string :only [lower-case]]))

(defn to-words [sentence]
  (map lower-case (re-seq #"\w+" sentence))
)

(defn word-count [sentence]
  (->> sentence (to-words) (frequencies))
)

File: python/word-count/word_count.py
import re

def word_count(sentence):
    count = {}
    words = re.split(r'[_\W]', sentence.lower())
    for word in words:
        if word.isalnum():
            if word not in count.keys():
                count[word] = 1
            else:
                count[word] = count[word] + 1
    return count
